src/NUnitFramework/tests/Api/FrameworkControllerTests.cs

```diff
@@ -132,63 +132,89 @@ public void InsertSettingsElement_SettingIsValue_CreatesASettingElementPerKey()
             var settingNode = inserted.FirstChild;
-            
-            Assert.That(2, Is.EqualTo(settingNode.ChildNodes.Count));
+
+            Assert.That(settingNode.ChildNodes.Count, Is.EqualTo(2));
         }
 
-        [Test]
-        public void InsertSettingsElement_SettingIsDictionary_CreatesEntriesWithKeysAndValuesFromDictionary()
+        [TestCaseSource(nameof(SettingsData))]
+        public void InsertSettingsElement_SettingIsDictionary_CreatesValueAttributeForDictionaryElements(string value)
+        {
+            var outerNode = new TNode("test");
+            var testSettings = new Dictionary<string, object>
+            {
+                ["outerkey"] = new Dictionary<string, object> { { "key1", "value1" }, { "key2", value } }
+            };
+
+            var inserted = FrameworkController.InsertSettingsElement(outerNode, testSettings);
+            var settingNode = inserted.FirstChild;
+
+            Assert.That(settingNode.ChildNodes.Count, Is.EqualTo(2));
+            Assert.That(settingNode.Attributes["value"], Is.EqualTo($"[key1, value1], [key2, {value}]"));
+        }
+
+        [TestCaseSource(nameof(SettingsData))]
+        public void InsertSettingsElement_SettingIsDictionary_CreatesEntriesWithKeysAndValuesFromDictionary(string value)
         {
             var outerNode = new TNode("test");
             var testSettings = new Dictionary<string, object>();
-            testSettings.Add("outerkey", new Dictionary<string, object> { { "key1", "value1" }, { "key2", "value2" } });
+            testSettings.Add("outerkey", new Dictionary<string, object> { { "key1", "value1" }, { "key2", value } });
 
             var inserted = FrameworkController.InsertSettingsElement(outerNode, testSettings);
             var settingNode = inserted.FirstChild;
 
             var key1Node = settingNode.ChildNodes[0];
-            Assert.That("key1", Is.EqualTo(key1Node.Attributes["key"]));
-            Assert.That("value1", Is.EqualTo(key1Node.Attributes["value"]));
+            Assert.That(key1Node.Attributes["key"], Is.EqualTo("key1"));
+            Assert.That(key1Node.Attributes["value"], Is.EqualTo("value1"));
 
             var key2Node = settingNode.ChildNodes[1];
-            Assert.That("key2", Is.EqualTo(key2Node.Attributes["key"]));
-            Assert.That("value2", Is.EqualTo(key2Node.Attributes["value"]));
+            Assert.That(key2Node.Attributes["key"], Is.EqualTo("key2"));
+            Assert.That(key2Node.Attributes["value"], Is.EqualTo(value));
+        }
+
+        public static IEnumerable SettingsData()
+        {
+            yield return new TestCaseData("value");
+            yield return new TestCaseData("");
+            yield return new TestCaseData("<value>");
+            yield return new TestCaseData("\"value\"");
+            yield return new TestCaseData("'value'");
+            yield return new TestCaseData("value1;value2");
```

Comment:
I wanted to test values that might cause problems in an attribute. There was nothing to worry about, but it gave me a bit of confidence in the change 😄